### PR TITLE
Unified handling for unstable command line flags across compiler and driver

### PR DESCRIPTION
**Pull request by JustusAdam**
### Description of changes: 

Currently the compiler and driver use two separate representations for unstable features. This unifies the handling so that both use the same mechanism.

It also centralizes the authority for parsing and serializing those representations to one place so that both individual features and collections of them can be converted to and from their respective command line form.

### Resolved issues:

Resolves #ISSUE-NUMBER

### Related RFC:

Optional #ISSUE-NUMBER.

### Testing:

* How is this change tested? Tested by existing test cases

* Is this a refactor change? Yes

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Comment by jaisnan** [comment]
Taking a look

File: kani_metadata/src/unstable.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//! Unified handling of unstable feature flags across Kani.
```
Comment (feliperodri) [inline kani_metadata/src/unstable.rs:3]:
```suggestion
//! Unified handling of unstable-feature flags across Kani.
```

File: kani_metadata/src/unstable.rs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    Stubbing,
+    /// Generate a C-like file equivalent to input program used for debugging purpose.
+    GenC,
+    /// Allow Kani to link against C code.
+    CFfi,
+    /// Enable concrete playback flow.
+    ConcretePlayback,
+    /// Enable Kani's unstable async library.
+    AsyncLib,
+    /// Enable line coverage instrumentation/reports
```
Comment (feliperodri) [inline kani_metadata/src/unstable.rs:81]:
```suggestion
    /// Enable line coverage instrumentation/reports.
```

File: kani_metadata/src/unstable.rs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+#[derive(clap::Args, Debug)]
+pub struct EnabledUnstableFeatures {
+    #[clap(short = 'Z', long = "unstable", num_args(1), value_name = "UNSTABLE_FEATURE")]
+    enabled_unstable_features: Vec<UnstableFeature>,
+}
+
+impl EnabledUnstableFeatures {
+    /// The preferred way to serialize these unstable features back into a
+    /// format that can be used as command line arguments fo an invocation of
+    /// e.g. the compiler
```
Comment (feliperodri) [inline kani_metadata/src/unstable.rs:108]:
```suggestion
    /// e.g. the compiler.
```

File: kani_metadata/src/unstable.rs [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    ///
+    /// See also the [module level documentation][self]
```
Comment (feliperodri) [inline kani_metadata/src/unstable.rs:110]:
```suggestion
    /// See also the [module level documentation][self].
```

**Review by feliperodri — APPROVED** [review]
Thank you! Only minor comments, please address them before merging.

**Review by jaisnan — APPROVED** [review]

**Pull request merged**